src/draw_target.rs

```diff
@@ -350,7 +351,15 @@ impl RateLimiter {
 
         // We add `new` to `capacity`, subtract one for returning `true` from here,
         // then make sure it does not exceed a maximum of `MAX_BURST`, then store it.
-        self.capacity = Ord::min(MAX_BURST, self.capacity + new as u8 - 1);
+        self.capacity = Ord::min(
```

Comment:
This is indeed pretty ugly, and IMO we can simplify this. Avoiding `as` conversions makes sense for fallible conversions but not for infallible ones. We can also define `MAX_BURST` as `u128` if that makes life easier. I don't like the use of `clamp()` with 0 as the floor since we're operating in the unsigned domain. That would get us this:

```rust
// Invariant: `new` must be > 1, otherwise we would have returned above 
self.capacity = Ord::min(self.capacity as u128 + new - 1, MAX_BURST) as u8;
```